doc/modules/impute.rst

```diff
@@ -120,3 +120,47 @@ whether or not they contain missing values::
          [False,  True, False, False]])
   >>> indicator.features_
   array([0, 1, 2, 3])
+
+When using it in a pipeline, be sure to use the :class:`FeatureUnion` to add
+the indicator features to the regular features. First we obtain the `annael`
+dataset from OpenML, which has many missing values.
+
+  >>> import sklearn.datasets
+  >>> import sklearn.impute
+  >>> import sklearn.model_selection
+  >>> import sklearn.pipeline
+  >>> import sklearn.tree
+  >>> X, y = sklearn.datasets.fetch_openml('anneal', 1, return_X_y=True)
+  >>> X_train, X_test, y_train, _ = sklearn.model_selection.train_test_split(X, y, test_size=100, random_state=0)
+
+Now we create a :class:`FeatureUnion`. All features will be imputed using
+:class:`SimpleImputer`, in order to enable classifiers to work with this data.
+Additionally, it adds the the indicator variables from
+:class:`MissingIndicator`.
+
+  >>> transformer = sklearn.pipeline.FeatureUnion(
+  ...     transformer_list=[
+  ...         ('features', sklearn.impute.SimpleImputer(strategy='mean')),
+  ...         ('indicaters', sklearn.impute.MissingIndicator(features='all'))])
+  >>> clf = sklearn.pipeline.make_pipeline(transformer,
+  ...                                      sklearn.tree.DecisionTreeClassifier())
+
+
+Note that the anneal dataset has 38 columns. By applying the `features='all'`
```

Comment:
I don't think this section provides very much value.

Reply:
My motivation was that it adds an explanation why we can expect the number of columns that we see (38*2=76). I can remove this of course, if you think this is common knowledge. WDYT?

Reply:
Note that I changed the dataset to audiology. anneal has columns that are completely empty, which are silently dropped by Simple Imputer. This would make the description a bit more complicated.